ISPN-9350 Dynamic parser versions

tristantarrant opened the pull request:
https://issues.jboss.org/browse/ISPN-9350
https://issues.jboss.org/browse/ISPN-9364

ISPN-9350 Dynamic parser versions
ISPN-9364 Forward-port of Data Grid schema support

tristantarrant commented:
added 8.3, 8.4 and 8.5 schema support

danberindei commented on Jenkinsfile-release:
```diff
@@ -6,7 +6,10 @@ pipeline {
     parameters {
-        string(defaultValue: '0.0.0.Qualifier', description: 'Release version', name: 'version')
+        string(name: 'version', defaultValue: '0.0.0.Qualifier', description: 'Release version')
+        string(name: 'nextVersion', defaultValue: '', description: 'Next releasae (blank to stay on current SNAPSHOT)')
```
Typo: releasae

tristantarrant replied:
I'll remove the Jenkinsfile-release as it doesn't work properly still.

danberindei commented on core/src/test/resources/configs/unified/8.3.xml:
```diff
@@ -0,0 +1,262 @@
+<infinispan
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xsi:schemaLocation="urn:infinispan:config:8.3 http://www.infinispan.org/schemas/infinispan-config-8.3.xsd"
+      xmlns="urn:infinispan:config:8.3">
```
Did you intend to add these 8.x.xml files, or did they slip in when testing the script?

tristantarrant replied:
It is intentional

danberindei commented on server/integration/infinispan/src/test/java/org/jboss/as/clustering/infinispan/subsystem/SubsystemParsingTestCase.java:
```diff
@@ -52,36 +59,35 @@
+            }
+            data.add(new Object[]{xmlPath, properties});
+        }
+        return data;
```
Here too, I'd like a check that the xml for the current version exists.

danberindei commented on server/integration/infinispan/src/test/resources/org/jboss/as/clustering/infinispan/subsystem/subsystem-infinispan_8_3.xml:
```diff
@@ -0,0 +1,201 @@
+<subsystem xmlns="urn:infinispan:server:core:8.3">
```
Some more 8.x XMLs and properties here...

tristantarrant commented on core/src/test/java/org/infinispan/configuration/parsing/LegacyXmlFileParsingTest.java:
```diff
@@ -0,0 +1,26 @@
+   @Test(expectedExceptions = CacheConfigurationException.class)
+   public void testUnsupportedConfiguration() throws Exception {
+      withCacheManager(new CacheManagerCallable(
+            TestCacheManagerFactory.fromXml("configs/legacy/6.0.xml", true)) {
```
ok

tristantarrant replied:
This will actually allows us to properly validate some config which would otherwise be overridden by TestCacheManagerFactory.

tristantarrant commented on core/src/test/java/org/infinispan/configuration/parsing/UnifiedXmlFileParsingTest.java:
```diff
@@ -69,515 +72,531 @@
+      for (int i = 0; i < paths.size(); i++) {
+         configurationFiles[i] = new Object[]{paths.get(i)};
+      }
+      return configurationFiles;
```
ok

tristantarrant commented:
@danberindei I've fixed everything apart from extracting the methods from the enum :)
Also, while I haven't added reversioning to the Jenkinsfile-release file, I have fixed a bug which pushed the modified poms to master, instead of just putting them in the tag

danberindei commented on core/src/test/java/org/infinispan/configuration/parsing/UnifiedXmlFileParsingTest.java:
```diff
@@ -69,515 +72,531 @@
+   public enum ParserVersionCheck {
+      INFINISPAN_93(9, 3) {
+         @Override
+         public void check(EmbeddedCacheManager cm) {
```
I'm not a fan of huge anonymous classes, even (or especially) when they're declared as enum values. In general I as little whitespace to the left of my code as possible :)

tristantarrant replied:
So no forgiveness, even for a test ?

danberindei commented:
Integrated, thanks Tristan!
